lib/utapi/utilities.js

```diff
@@ -182,20 +182,28 @@ export function listMetrics(metricType) {
  * @return {function} - `utapi.pushMetric`
  */
 export function pushMetric(action, log, metricObj) {
-    let accountId = undefined;
-    // If `authInfo` is included by the API, get the account's canonical ID for
-    // account-level metrics.
-    if (metricObj.authInfo) {
-        accountId = metricObj.authInfo.getCanonicalID();
-    }
-    const { bucket, byteLength, newByteLength, oldByteLength,
-        numberOfObjects } = metricObj;
-    return utapi.pushMetric(action, log.getSerializedUids(), {
+    const { bucket, byteLength, newByteLength, oldByteLength, numberOfObjects,
+        authInfo } = metricObj;
+    const utapiObj = {
         bucket,
-        accountId,
         byteLength,
         newByteLength,
         oldByteLength,
         numberOfObjects,
-    });
+    };
+    // If `authInfo` is included by the API, get the account's canonical ID for
+    // account-level metrics and the shortId for user-level metrics.
+    if (authInfo) {
+        utapiObj.accountId = authInfo.getCanonicalID();
+        const arn = authInfo.getArn();
+        const shortId = authInfo.getShortid();
+        if (arn && shortId) {
```

Comment:
why not use the isRequesterAnIAMUser method to determine if the authInfo relates to a user?